core/src/sumeragi/view_change.rs

```diff
@@ -204,3 +212,69 @@ impl ProofChain {
         Ok(())
     }
 }
+
+mod candidate {
+    use indexmap::IndexSet;
+    use parity_scale_codec::Input;
+
+    use super::*;
+
+    #[derive(Decode)]
+    struct SignedProofCandidate {
+        signatures: Vec<ViewChangeProofSignature>,
+        payload: ViewChangeProofPayload,
+    }
+
+    impl SignedProofCandidate {
+        fn validate(self) -> Result<SignedViewChangeProof, &'static str> {
+            self.validate_signatures()?;
+
+            self.signatures
+                .iter()
+                .map(|signature| &signature.0)
+                .try_fold(IndexSet::new(), |mut acc, elem| {
+                    if !acc.insert(elem) {
+                        return Err("Duplicate signature in proof");
+                    }
+
+                    Ok(acc)
+                })?;
```

Comment:
The same as . Also, boundary between `validate` and `validate_signatures` is inconsistent with that of `SignedBlockCandidate`

Reply:
it's different because we have to compare every signature to every other. Collecting into a set is the easiest